deltacat/utils/pyarrow.py

```diff
@@ -167,6 +167,29 @@ def _get_kwargs(
         return kwargs
 
 
+class ReadKwargsProviderPyArrowSchemaOverride(ContentTypeKwargsProvider):
+    """ReadKwargsProvider impl that explicitly maps column names to column types when loading
+    delimited text files into a PyArrow table. Disables the default type inference behavior on the defined columns.
+    """
+    def __init__(self, schema: Optional[pa.Schema] = None):
+        self.schema = schema
+
+    def _get_kwargs(
+            self,
+            content_type: str,
+            kwargs: Dict[str, Any]) -> Dict[str, Any]:
+        if content_type in DELIMITED_TEXT_CONTENT_TYPES:
+            convert_options = kwargs.get("convert_options", pacsv.ConvertOptions())
+            if self.schema:
+                convert_options.column_types = self.schema
+            kwargs["convert_options"] = convert_options
+        elif content_type == ContentType.PARQUET:
+            # kwargs here are passed into `pyarrow.parquet.read_table`. Only supported in PyArrow 8.0.0+
```

Comment:
We should reflect this dependency on PyArrow 8 in `setup.py` and `requirements.txt`. Unfortunately, it seems like Ray Datasets is still on Ray 7...

Reply:
We should, my local changes did have them set to `pyarrow == 8.0.0`. I actually ended up needing this change to compact successfully; the dedupe step downloads hash bucket entries from previous compaction rounds in PARQUET, so they would also be loaded into PyArrow with inconsistent schemas / column ordering too.